app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,39 @@
+class ReviewsController < ApplicationController
+  before_action :set_user
+
+  def index
+    if params[:query].present?
+      author = User.where("name ILIKE ?", "%#{params[:query]}%").first
+      @reviews = Review.where(receiver: @user, author: author)
+    else
+      @reviews = Review.where(receiver: @user)
```

Comment:
This is the only line we need here (no need for line 5-8), because we will not search for reviews, but only get the reviews associated to a certain user